server/api/src/rsGetHostForGet.cpp

```diff
@@ -17,8 +17,73 @@
 #include "irods/rsGetHostForGet.hpp"
 #include "irods/irods_resource_backport.hpp"
 #include "irods/irods_resource_redirect.hpp"
+#include "irods/irods_at_scope_exit.hpp"
 
+#include <nlohmann/json.hpp>
+
+#include <set>
 #include <cstring>
+#include <optional>
+
+int rs_get_resource_info_for_operation(
+    rsComm_t*     rsComm,
+    dataObjInp_t* dataObjInp,
+    char **       outInfo )
+{
+    rodsServerHost_t *rodsServerHost;
+    const int remoteFlag = getAndConnRemoteZone(rsComm, dataObjInp, &rodsServerHost, REMOTE_OPEN);
+    if (remoteFlag < 0) {
+        return remoteFlag;
+    }
+    else if (REMOTE_HOST == remoteFlag) {
+        const int status = rc_get_resource_info_for_operation(rodsServerHost->conn, dataObjInp, outInfo);
+        if (status < 0) {
+            return status;
+        }
+    }
+    else {
+        std::string location {};
+        // =-=-=-=-=-=-=-
+        // working on the "home zone", determine if we need to redirect to a different
+        // server in this zone for this operation.  if there is a RESC_HIER_STR_KW then
+        // we know that the redirection decision has already been made
+        std::string hier{};
+        const char * op_type = getValByKey(&dataObjInp->condInput, GET_RESOURCE_INFO_OP_TYPE_KW);
+        std::set<std::string> allowed_ops{ irods::CREATE_OPERATION,
+                                           irods::WRITE_OPERATION,
+                                           irods::OPEN_OPERATION  };
```

Comment:
Could these be const refs? i.e.
```suggestion
        const std::set<const std::string&> allowed_ops{ irods::CREATE_OPERATION,
                                                        irods::WRITE_OPERATION,
                                                        irods::OPEN_OPERATION  };
```

Reply:
maybe

Reply:
changed to const